### ISSUE TITLE
Array becomes not equal to itself

### ISSUE THREAD

**Issue by danieldegtyarev** [bridgedotnet/Bridge#3222]
See repro here https://deck.net/4f3cf55d07690045bf3c254690074268

```
interface IProperty{
    object Value {get;set;}
}
interface IProperty<T> : IProperty
{
	T Value{get;set;}
}
class Property<T> : IProperty<T>
{
    T value;
    public T Value{
    get{return value;}
    set{this.value = value;}
    }
    
    object IProperty.Value{
    get{return value;}
    set{this.value = (T)value;}
    }
}

public class Program
{
    public static void Main()
    {
        var array = new [] { "abc", "def" };
        IProperty p1 = new Property<string[]>();
        p1.Value = array;
                
        Console.WriteLine(Object.Equals(array, p1.Value));        
    }
}```

**Comment by geoffreymcgill** [comment]
Thanks for the report. I have reproduced the defect. 

https://dotnetfiddle.net/32PV2c

We will investigate and fix right away.

**Comment by fabriciomurta** [comment]
The fix has just been merged upstream, it made it to the to-be-released Bridge 16.5.0!
